src/SymEngine.jl

```diff
@@ -149,3 +149,14 @@ show(io::IO, b::Basic) = print(io, toString(b))
 
 end
 
+abstract Basic
+
```

Comment:
There are some classes in SymEngine where it is a concrete type and there are subclasses of it as well.
You should take a look at how other libraries have done this.